### PR TITLE
Convert more code to use new time classes

### PR DESCRIPTION
**Pull request by BevapDin**
Beside changing the type of some variables to `time_point` / `time_duration`, this changes `calendar::print_time`:
- renamed to `to_string` (as it does not print like the curses functions).
- moved the part that was printing in special format into "weather.cpp", as it's only used there.

### CODE REVIEW

**Comment by ZhilkinSerg** [comment]
Something is bad with time display, when using any watches:

# game start:

image

# 3 steps north:

image

# more steps:

[image](https://user-images.githubusercontent.com/16213433/38163293-7a11f290-34fa-11e8-891a-cc8d0a65e171.png)

image

**Comment by ZhilkinSerg** [comment]
Seconds are now displayed as multiples of 10, but there were multiples of 6 previously:

image

Was it intended?

**Comment by BevapDin** [comment]
> Was it intended?

It wasn't. I'm switching between two notebooks, one older than the other (a full build takes some hours), and was mixing up the branches.

File: src/weather_gen.cpp [hunk 1]
```diff
@@ -42,8 +42,7 @@ w_point weather_generator::get_weather( const tripoint &location, const calendar
     double A( raw_noise_4d( x, y, z, modSEED ) * 8.0 );
     double W;
 
-    const double now( double( t.turn_of_year() + to_turns<int>( calendar::season_length() ) / 2 ) /
-                      to_turns<int>( calendar::year_length() ) ); // [0,1)
+    const double now( time_past_new_year( t ) / calendar::year_length() ); // [0,1)
```
Comment (ZhilkinSerg) [inline src/weather_gen.cpp]: @BevapDin: People say weather became somewhat colder (https://discourse.cataclysmdda.org/t/15260) and this line is likely candidate.

Shouldn't it be `const double now( ( time_past_new_year( t ) + calendar::season_length() / 2 ) / calendar::year_length() )`, actually?

**Pull request merged**